pkg/provisioning/config/yaml/v2/model.go

```diff
@@ -64,6 +89,7 @@ type Connector struct {
 type Processor struct {
 	ID        string            `yaml:"id" json:"id"`
 	Type      string            `yaml:"type" json:"type"`
```

Comment:
We should mark this field with `omitempty` so it's ignored when exporting the config.
```suggestion
	Type      string            `yaml:"type,omitempty" json:"type,omitempty"`
```

Reply:
I was thinking about that too today, but noticed we don't that for some other fields (such as condition, workers, which are optional). Any reason not  to add `omitempty` to those too?